src/chapter08-porting-wolfHSM.md

```diff
@@ -1 +1,153 @@
+
+## WolfHSM Porting Interface
+
+Ports must implement hardware-specific interfaces:
+- Non-volatile memory
+- Flash-like devices (read, erase, program, blankcheck, etc)
+
+Crypto Hardware
+- TRNG, Keys, symmetric/asymmetric crypto
+
+Platform Interface
+- Boot sequence, application core reset, memory limitations
+- Port and configuration are selected at compile time
+
+# General Porting Configuration 
+
+## WolfHSM Port Configuration Example
+
+Flash Interface using Posix flash file simulator
+
+```c
+#include “wolfhsm/wh_nvm_flash.h”								/*Server NVM Manager using flash */
+#include “wolfhsm/wh_flash.h”                  /*Abstract Flash API */
+#include “port/posix/posix_flash_file.h”       /*Port Implementation of Flash API */
+
+/* Specify flash callback functions */
+const whFlashCb pff_cb[1] = { POSIX_FLASH_FILE_CB };
+
+/* Allocate implementation-specific context with empty initializer */
+posixFlashFileContext pff_context[1] = {0};
+
+/* Specify implementation-dependent configuration */
+posixFlashFileConfig pff_cfg[1] = {{
+	.filename       = “my_flash_file.bin”,
+	.partition_size = 16384,
+	.erased_byte    = 0xFF,
+}};
+
+/* Connect NVM Flash layer to this PFF instance */
+const whNvmFlashConfig nvm_flash_config[1] = {{
+	.cb      = pff_cb,
+	.context = (void*)pff_context,
+	.config  = (const void*)pff_config,
+}};
+```
+
+# WolfHSM Posix Porting Example
+
+## General Steps for Porting Configuration 
+
+## Porting Configuration Example
+
+# WolfHSM Porting Skeleton 
+
+## General Steps for Porting Configuration 
+
+## Porting Configuration Example
+
+# WolfHSM Porting Infineon Aurix TC3xx
+
+## General Steps for Porting Configuration 
+
+## Porting Configuration Example
+
+# WolfHSM Porting ST SPC58NN
+
+## General Steps for Porting Configuration 
+
+## Porting Configuration Example
+
+# Summary
+
+* Each of the implemented port-specific code and resources are kept in port directories organized first by platform vendor and possibly further by product.  Each of the ports is expected to provide the glue logic between wolfHSM abstractions (transport, NVM objects, flash, and cryptocb) and the native or vendor-provided libraries.
+
+Due to the sensitive nature of some platform code, not all of the source and glue logic can be provided in this public repo, but the base directories of these ports are listed here with any public interfaces that can be provided and additional contact information.
```

Comment:
I'd nix all this. It doesn't make sense to talk about the existing ports that we have, since users will not need to do any work to port to this platform. This guide was more meant to explain how to port to a platform that we don't already have a port for. It is probably appropriate to just point to the POSIX port for examples of how you do all the port stuff. If you want you can walk them through how it works, but probably overkill for now.